Firestore/Source/Remote/FSTStream.mm

```diff
@@ -251,18 +253,17 @@ - (void)startWithDelegate:(id)delegate {
   FSTAssert(_delegate == nil, @"Delegate must be nil");
   _delegate = delegate;
 
-  [self.credentials
-      getTokenForcingRefresh:NO
-                  completion:^(FSTGetTokenResult *_Nullable result, NSError *_Nullable error) {
-                    error = [FSTDatastore firestoreErrorForError:error];
-                    [self.workerDispatchQueue dispatchAsyncAllowingSameQueue:^{
-                      [self resumeStartWithToken:result error:error];
-                    }];
-                  }];
+  [self.credentials getTokenForcingRefresh:NO
+                                completion:^(const Token &result, NSError *_Nullable error) {
+                                  error = [FSTDatastore firestoreErrorForError:error];
+                                  [self.workerDispatchQueue dispatchAsyncAllowingSameQueue:^{
+                                    [self resumeStartWithToken:result error:error];
```

Comment:
This doesn't work. I'm surprised this isn't crashing you.

The token passed into the completion handler is a stack allocated temporary whose reference you're capturing to pass into the block passed into the dispatch queue. By the time that block actually runs the original token will have been destroyed.

You need to capture this by value rather than by reference to avoid these lifetime issues.